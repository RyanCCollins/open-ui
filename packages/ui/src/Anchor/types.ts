import { Props } from './';

export { Props };
