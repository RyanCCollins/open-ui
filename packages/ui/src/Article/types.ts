import BoxProps from '../Box/types';

export interface ArticleProps extends BoxProps {
  content?: string;
  children?: JSX.Element;
}
