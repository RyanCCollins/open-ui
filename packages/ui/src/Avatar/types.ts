import * as React from 'react';
import Avatar from './';

export type ImageSize = 'thumb' | 'small' | 'medium' | 'large';
export interface Props extends React.Props<typeof Avatar> {
  src: string;
  name?: string;
  size?: ImageSize;
}
