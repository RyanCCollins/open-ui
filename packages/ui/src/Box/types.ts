import * as React from 'react';
import Box from './';

export type Size = 'none' | 'small' | 'medium' | 'large' | 'xlarge';
export interface SizeObject { horizontal?: Size; vertical?: Size; };
export type Breakpoint = 'mobile' | 'tablet' | 'desktop';
export interface ResponsiveSize {
  mobile?: Size | SizeObject,
  tablet?: Size | SizeObject,
  desktop?: Size | SizeObject,
}
export type BoxSize =  'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'xxlarge' | 'full';
export interface BoxSizeObject { horizontal?: BoxSize; vertical?: BoxSize; };
export interface FullObject { horizontal: boolean; vertical: boolean; };
export type Full = boolean | FullObject;
export type WrapOption = 'wrap' | 'wrap-reverse' | 'nowrap';
export type Rem = number;
export type Px = number;

export interface SizeStyle {
  height: string;
  width: string;
}

interface BoxProps extends React.HTMLProps<typeof Box> {
  alignItems?: 'flex-start' | 'center' | 'flex-end' | 'baseline' | 'stretch';
  flexDirection?: 'row' | 'column';
  justifyContent?: 'flex-start' | 'center' | 'flex-end' | 'space-between' | 'space-around';
  flexWrap?: boolean;
  reverse?: boolean;
  boxSize?: BoxSize | BoxSizeObject;
  pad?: Size | SizeObject;
  children?: JSX.Element;
  margin?: Size | SizeObject;
  backgroundColor?: string;
  backgroundImage?: string;
  full?: boolean | { horizontal?: boolean, vertical?: boolean };
  selectable?: boolean;
};

export default BoxProps;
