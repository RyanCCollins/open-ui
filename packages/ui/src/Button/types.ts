import { Props } from './';

export { Props };
export type Size = 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
