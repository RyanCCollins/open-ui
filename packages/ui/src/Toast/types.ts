import { Props } from './';

export { Props };

export type Status = 'ok' | 'warning' | 'error' | 'none';
