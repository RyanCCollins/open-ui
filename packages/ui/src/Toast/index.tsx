import * as React from 'react';
import Box from '../Box';
import Component, { Button } from './styles';
import BoxProps from '../Box/types';
import { Status } from './types';

export interface Props extends BoxProps {
  message?: JSX.Element;
  children?: JSX.Element;
  isVisible?: boolean;
  onClose?: () => void;
  status?: Status;
}

export interface State {
  isVisible: boolean;
  isUnmounting: boolean;
}

export default class Toast extends React.Component<Props, State> {
  public static defaultProps = {
    status: 'none' as Status,
  };
  constructor() {
    super();
    this.handleClose = this.handleClose.bind(this);
    this.state = {
      isVisible: true,
      isUnmounting: false,
    };
  }
  public componentDidMount() {
    setTimeout(() => {
      this.handleClose();
    }, 5000);
  }
  private handleClose() {
    this.setState({
      isUnmounting: true,
    });
    setTimeout(() => {
      if (this.props.onClose) {
        this.props.onClose();
      }
      this.setState({
        isUnmounting: false,
        isVisible: false,
      });
    }, 1000);
  }
  public render() {
    const { isVisible } = this.state;
    const { message, children, status } = this.props;
    if (!isVisible) {
      return null;
    }
    return (
      <Component status={status} {...this.state}>
        <Box
          alignItems="center"
          justifyContent="center"
          flexDirection="row"
          size={60}
          pad="medium"
          style={{ width: '100vw' }}
        >
          <Box style={{ flexGrow: 2 }}>
            {message && message}
            {children && children}
          </Box>
          <Box alignItems="center" justifyContent="center">
            <Button onClick={this.handleClose}>
              âœ•
            </Button>
          </Box>
        </Box>
      </Component>
    );
  }
}
