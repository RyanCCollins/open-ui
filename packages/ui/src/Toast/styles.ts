const styled = require('styled-components').default;
const { keyframes, css } = require('styled-components');
import { Status } from './types';
import colorMap from '../theming/colorMap';

const backgroundColor = (status: Status) => css`
  background-color: ${colorMap[status] || colorMap.offwhite};
`;

const unmount = keyframes`
  0% { transform: translateY(0%); }
  100% { transform: translateY(-100%); }
`;

const mount = keyframes`
  0% { transform: translateY(-100%); }
  100% { transform: translateY(0%); }
`;

export const Button = styled.button`
  background-color: transparent;
  border: 0px;
  color: #333;
  cursor: pointer;
  font-size: 2rem;
`;

export default styled.div`
  position: fixed;
  font-size: 1.3rem;
  top: 0;
  left: 0;
  right: 0;
  z-index: 100;
  display: flex;
  flex-direction: row;
  ${(props) => backgroundColor(props.status)};
  animation: ${(props) => props.isUnmounting ? unmount : mount} 1s;
`;
