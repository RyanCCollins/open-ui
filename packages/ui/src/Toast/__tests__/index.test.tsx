import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Toast from '../';
import { Status } from '../types';
import props from './__mocks__/toast.mock';

describe('<Toast />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Toast>
        Jello
      </Toast>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with different props', () => {
    const wrapper = shallow(
      <Toast {...props} status={props.status as Status}>
        Foobar
      </Toast>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
