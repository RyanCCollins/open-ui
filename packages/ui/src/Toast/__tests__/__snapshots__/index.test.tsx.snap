// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> should render with default props 1`] = `
<styled.div
  isUnmounting={false}
  isVisible={true}
  status="none"
>
  <Box
    alignItems="center"
    flexDirection="row"
    justifyContent="center"
    pad="medium"
    size={60}
    style={
      Object {
        "width": "100vw",
      }
    }
  >
    <Box
      style={
        Object {
          "flexGrow": 2,
        }
      }
    >
      Jello
    </Box>
    <Box
      alignItems="center"
      justifyContent="center"
    >
      <styled.button
        onClick={[Function]}
      >
        ✕
      </styled.button>
    </Box>
  </Box>
</styled.div>
`;

exports[`<Toast /> should render with different props 1`] = `
<styled.div
  isUnmounting={false}
  isVisible={true}
  status="ok"
>
  <Box
    alignItems="center"
    flexDirection="row"
    justifyContent="center"
    pad="medium"
    size={60}
    style={
      Object {
        "width": "100vw",
      }
    }
  >
    <Box
      style={
        Object {
          "flexGrow": 2,
        }
      }
    >
      Foobar
    </Box>
    <Box
      alignItems="center"
      justifyContent="center"
    >
      <styled.button
        onClick={[Function]}
      >
        ✕
      </styled.button>
    </Box>
  </Box>
</styled.div>
`;
